packages/headless/src/controllers/insight/smart-snippet/headless-insight-smart-snippet-interactive-inline-links.ts

```diff
@@ -0,0 +1,164 @@
+import {InsightEngine} from '../../../app/insight-engine/insight-engine';
+import {questionAnswering, search} from '../../../app/reducers';
+import {
+  logOpenSmartSnippetInlineLink,
+  logOpenSmartSnippetSuggestionInlineLink,
+} from '../../../features/question-answering/question-answering-insight-analytics-actions';
+import {
+  QuestionAnsweringSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {getObjectHash} from '../../../utils/utils';
+import {
+  buildInteractiveResultCore,
+  InteractiveResultCore,
+} from '../../core/interactive-result/headless-core-interactive-result';
+import {InlineLink} from '../../smart-snippet/headless-smart-snippet-interactive-inline-links';
+
+export interface SmartSnippetInteractiveInlineLinksOptions {
+  /**
+   * The amount of time to wait before selecting the result after calling `beginDelayedSelect`.
+   *
+   * @defaultValue `1000`
+   */
+  selectionDelay?: number;
+}
+
+export interface SmartSnippetInteractiveInlineLinksProps {
+  /**
+   * The options for the result controller core.
+   */
+  options?: SmartSnippetInteractiveInlineLinksOptions;
+}
+
+export interface SmartSnippetInteractiveInlineLinks {
+  /**
+   * Selects the result, logging a UA event to the Coveo Platform if the result wasn't selected before.
+   *
+   * In a DOM context, it's recommended to call this method on all of the following events:
+   * * `contextmenu`
+   * * `click`
+   * * `mouseup`
+   * * `mousedown`
+   */
+  selectInlineLink(link: InlineLink, questionAnswerId?: string): void;
+  /**
+   * Prepares to select the result after a certain delay, sending analytics if it was never selected before.
+   *
+   * In a DOM context, it's recommended to call this method on the `touchstart` event.
+   */
+  beginDelayedSelectInlineLink(
+    link: InlineLink,
+    questionAnswerId?: string
+  ): void;
+  /**
+   * Cancels the pending selection caused by `beginDelayedSelect`.
+   *
+   * In a DOM context, it's recommended to call this method on the `touchend` event.
+   */
+  cancelPendingSelectInlineLink(
+    link: InlineLink,
+    questionAnswerId?: string
+  ): void;
+}
+
+/**
+ * Creates the insight result controller for SmartSnippet.
```

Comment:
the search version doesn't document these marks all of these as @internal so they won't be publicly documented. Do we have good reason to go a different way for insight?